docfx/doxygen2markdown.cc

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "docfx/doxygen2markdown.h"
+#include <iostream>
+#include <sstream>
+#include <string_view>
+
+namespace {
+
+[[noreturn]] void UnknownChildType(std::string_view where,
+                                   pugi::xml_node const& child) {
+  std::ostringstream os;
+  os << "Unknown child in " << where << "(): name=" << child.name();
+  os << ", attributes=[";
+  for (auto const& a : child.attributes()) {
+    os << ", " << a.name() << "=" << a.as_string();
+  }
+  os << "]";
+  child.print(os);
+  throw std::runtime_error(std::move(os).str());
+}
+
+}  // namespace
+
+bool AppendIfPlainText(std::ostream& os, pugi::xml_node const& node) {
+  if (node.name() != std::string_view{} || !node.attributes().empty()) {
+    return false;
+  }
+  os << node.value();
+  return true;
+}
+
+bool AppendIfComputerOutput(std::ostream& os, pugi::xml_node const& node) {
+  if (node.name() != std::string_view{"computeroutput"}) return false;
```

Comment:
If you haven't already considered it, I'm wondering whether `std::string_view(node.name()) != "literal"` might be clearer than `node.name() != std::string_view("literal")`.  I'm not sure, so your choice.

Reply:
Good idea, done.